torch/fx/passes/shape_prop.py

```diff
@@ -35,11 +35,16 @@ def _extract_tensor_metadata(result : torch.Tensor) -> TensorMetadata:
     requires_grad = result.requires_grad
     stride = result.stride()
 
-    memory_formats = {
-        torch.contiguous_format,
-        torch.channels_last,
-        torch.channels_last_3d,
-    }
+    if torch._C._functorch.is_batchedtensor(result):
+        memory_formats = {
+            torch.contiguous_format,
+        }
```

Comment:
How is `_extract_tensor_metadata` used in shape prop?

Reply:
`extract_tensor_metadata` is invoked within the `CondHigherOrderVariable::call_function` to ensure the tensors returned by both the if and else branches have the same metadata. See:
https://github.com/pytorch/pytorch/blob/55ce4693ff2c0b6e50b8af323f36ecc7ff929638/torch/_dynamo/variables/higher_order_ops.py#L542-L560

Reply:
Okay this one's a bit trickier to resolve... cond needs to check that the result of true_fn and false_fn have the same properties. `_extract_tensor_metadata` is only meant to be used for actual concrete Tensors (and not Tensor subclasses). When we have vmap(torch.cond)(pred, true_fn, false_fn, x), what properties should we be checking? Should we check that true_fn(x) and false_fn(x) produces the same Tensor subclasses? (e.g. BatchedTensor vs regular Tensor)

cc @ydwu4

Reply:
I would be fine with us adding a `include_contiguity=True` flag to `_extract_tensor_metadata` and having the cond logic call extract_tensor_metadata(include_contiguity=False) to avoid this problem. Morally it seems like we only care about checking the metadata is the same when we're executing the cond with real non-subclass Tensors